### PR TITLE
Bluetooth: Shell: Add connect-name command

### PR DESCRIPTION
**Pull request by Thalley**
Add a new command, connect-name, which takes a string
as the single input and uses that to setup a scan filter
and connect to the first device that matches the name.

This command is thus basically shorthand for
bt scan-filter-set <name>
bt scan on
bt scan off
bt connect
bt scan-filter-clear-name

The idea of this command is to not only make it easier
find and connect to devices in a crowded environment,
but also to more easily connect to devices that 
changes their advertised address.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

### CODE REVIEW

**Comment by Thalley** [comment]
The main motive behind this PR is to make UPF testing easier :)

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/shell/bt.c [hunk 1]
```diff
@@ -206,12 +212,28 @@ static void scan_recv(const struct bt_le_scan_recv_info *info,
...
-#endif
+
+	/* Use the above auto_connect.addr address to automatically connect */
+	if (auto_connect.connect_name) {
+		cmd_scan_off(ctx_shell);
+
+		/* "connect" is what would be in argv[0] normally */
+		cmd_connect_le(ctx_shell, 1, (char *[]){ "connect" });
+	}
+#endif /* CONFIG_BT_CENTRAL */
```
[the remark this answers is not in the document]
Comment (Thalley) [reply subsys/bluetooth/shell/bt.c:223]: Good suggestion, thanks

**Review by cvinayak — DISMISSED** [review]

**Review by sjanc — DISMISSED** [review]
LGTM but I'd prefer to not suggest in commit message that privacy and fixed adv data (name here) is something that should be used :-)

**Comment by Thalley** [comment]
> LGTM but I'd prefer to not suggest in commit message that privacy and fixed adv data (name here) is something that should be used :-)

Removed the privacy reference :)

**Review by sjanc — APPROVED** [review]

**Review by MariuszSkamra — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**